portfolio-project-matching-app/backend/dao.js

```diff
@@ -465,18 +541,21 @@ const buildObjects = (coll, collSnap) => {
 
                     collSnap (collectionSnapshot): collectionSnapshot
 
+                    deep (boolean): indicates whether objects association
+                    properties should be populated. If no argument is entered,
+                    returned objects will be shallow (no association arrays
+                    populated)
+
     RETURN:         array of Project, User, or Technology objects
     */
-    // initialize array vessel to return to calling function
-    const objects = [];
-    // loop through collection snapshot docs, build object, add to array
-    for (const doc of collSnap.docs) {
-        objects.push(buildObject(coll, doc));
-    }
+    // build array of objects to return to the user
+    const objects =  await Promise.all(collSnap.docs.map((doc) => {
+        return buildObject(coll, doc, deep);
+    }));
     return objects;
 }
 
-const buildObject = (coll, doc) => {
+const buildObject = async (coll, doc, deep = false) => {
```

Comment:
Helper function to cut down on the number of dedicated functions for the different objects.